lib/haml/compiler.rb

```diff
@@ -277,7 +277,7 @@ def push_script(text, opts = {})
 
     def build_script_formatter(text, opts)
       text = "(#{text}).to_s"
-      if opts[:escape_html]
+      if opts[:escape_html] && !opts[:mime_type] == 'text/plain'
```

Comment:
I think it would be more clear to use the inequality operator here :blush:
```suggestion
      if opts[:escape_html] && opts[:mime_type] != 'text/plain'
```

Reply:
Thanks for spotting that. I'm not really sure why I wrote it that way. I've updated the PR.